### ISSUE TITLE
Pinata pin service integration defunct 

### ISSUE THREAD

**Issue by RubenKelevra** [ipfs/ipfs-desktop#1954]
- OS: Windows
 - Version of IPFS Desktop 0.18.1

**Describe the bug**
Pinata integration stopped working. So it worked before and after some update in the recent past it stopped working. I thought it's a temporary outage as I saw it first. But that's quite a while ago, maybe a month or two, and it's still not working.

**To Reproduce**
1. -> Settings
2. Click "+ Add Service"
3. Select Pinata
4. Go to Pinata's API page (which is linked)
5. Create a new key
6. Add it to the dialog opened in step 3
7. Click on Save
8. Pinata's status is "Pins: Failed to fetch"

**Expected behavior**
The integrated service connection should work.

**Screenshots**
![IPFS Desktop Pinata fail](https://user-images.githubusercontent.com/614929/149426619-c2b821f3-9f44-4163-a540-22a9b1e2af43.png)

**Comment by hacdias** [comment]
@RubenKelevra I think it's related to your API token: it's either expired or doesn't have the required permissions (Pinning API permissions).

**Comment by RubenKelevra** [comment]
> @RubenKelevra I think it's related to your API token: it's either expired or doesn't have the required permissions (Pinning API permissions).
> 
> ![Screenshot 2022-01-14 at 14 05 45](https://user-images.githubusercontent.com/5447088/149519919-d2b65e57-93b2-4dad-af64-8184b7c82c6a.png)

Well, the token is new and has all permissions active.

**Comment by hacdias** [comment]
Very odd, it works well here. Did you use the correct token? It's the JWT token (secret access token) and not the API secret.

![image](https://user-images.githubusercontent.com/5447088/149525024-f5ecb5bf-0703-4ba1-b08a-c1a582f37c0e.png)

**Comment by samuelsoderberg** [comment]
Yeah, I have the same problem.

It works intermittently. I got one file to Pinata, but not the other. And before I got the one file to work I had the same error message, "Failed to fetch".

<img width="1543" alt="image" src="https://user-images.githubusercontent.com/8748031/150012729-9936f3bb-5174-475d-9c27-842586449551.png">

Using the JWT with permissions, of course.

Possibly something wrong with Pinata?

**Comment by hacdias** [comment]
@RubenKelevra any updates on this?

**Comment by RubenKelevra** [comment]
Hey @hacdias, sorry for the delay. I think the issue was related to my ISP - it's working fine now without any change 🤷

Thanks for the efforts to help me, highly appreciated!

**Comment by lidel** [comment]
FWIW I noticed similar "false-negative" while testing webui in ipfs-desktop. 

Poked at things and this.. is not false-negative. 

Pinata is overloaded, and HTTP checks sent by `ipfs pin remote service ls --stat`  (used by ipfs-webui for determining if service is in a good shape) are not successful:

```console
$ ipfs pin remote service ls --stat                                                                                                         ~
Pinata       https://api.pinata.cloud/psa     invalid
estuary      https://api.estuary.tech/psa     invalid
pinata-debug http://127.0.0.1:5000/psa        invalid
pinata-test  https://testapi.pinata.cloud/psa invalid
```

I switched `estuary` to use valid endpoint and  calling `ipfs pin remote service ls --stat` multiple times  shows that the problem lies in Pinata throttling status check:

```console
$ ipfs pin remote service ls --stat                                                                                                         ~
Pinata       https://api.pinata.cloud/psa     0/0/11/0
estuary      https://api.estuary.tech/pinning 14/14/1/0
pinata-debug http://127.0.0.1:5000/psa        invalid
pinata-test  https://testapi.pinata.cloud/psa 0/0/16/5

$ ipfs pin remote service ls --stat                                                                                                         ~
Pinata       https://api.pinata.cloud/psa     invalid
estuary      https://api.estuary.tech/pinning 14/14/1/0
pinata-debug http://127.0.0.1:5000/psa        invalid
pinata-test  https://testapi.pinata.cloud/psa invalid
```

This can be reproduced with `curl` 

```console
$ curl -H "Authorization: Bearer secret" "https://api.pinata.cloud/psa/pins?limit=1&status=failed"
{"count":0,"results":[]}
$ curl -H "Authorization: Bearer secret" "https://api.pinata.cloud/psa/pins?limit=1&status=failed"
{"count":0,"results":[]}
$ curl -H "Authorization: Bearer secret" "https://api.pinata.cloud/psa/pins?limit=1&status=failed"
{"count":0,"results":[]}
$ curl -H "Authorization: Bearer secret" "https://api.pinata.cloud/psa/pins?limit=1&status=failed"
{"count":0,"results":[]}
$ curl -H "Authorization: Bearer secret" "https://api.pinata.cloud/psa/pins?limit=1&status=failed"
{"count":0,"results":[]}
$ curl -H "Authorization: Bearer secret" "https://api.pinata.cloud/psa/pins?limit=1&status=failed"
{"error":{"reason":"Rate limit exceeded","details":"You have exceeded the API rate limit. Wait a minute and slow down your requests."}}
```

@obo20 any suggestions how we avoid triggering this rate limiter?  
 `ipfs pin remote service ls --stat`  sends 4 requests (one for each status), and you seem to block around 6 requests in some short window – pretty low – perhaps we could just bump this to more reasonable throttling?

**Comment by RubenKelevra** [comment]
@lidel yeah, I can confirm that I still have issues. It jumped back to "Failed to fetch".

So I created a new key, and when I add it it shows that it's working and but it jumps to "Failed to fetch" when I turn on auto-upload.

So not sure if pinata doesn't want users to use their pin-service or if that's an IPFS issue but it's kind of annoying since I have to manually upload everything to their service.

**Comment by lidel** [comment]
I think we need to add support for `Retry-After: 60` to https://github.com/ipfs/go-pinning-service-http-client/

Pinata returns:
```
< HTTP/1.1 429 Too Many Requests
< Server: nginx/1.16.1
< Date: Mon, 07 Feb 2022 21:08:07 GMT
< Content-Type: application/json; charset=utf-8
< Content-Length: 135
< Connection: keep-alive
< X-Powered-By: Express
< Vary: Origin
< Access-Control-Allow-Credentials: true
< X-RateLimit-Limit: 5
< X-RateLimit-Remaining: 0
< X-RateLimit-Reset: 1644268110
< Retry-After: 60
```
